### ISSUE TITLE
Random segmentation fault while scanning

### ISSUE THREAD

**Issue by akarzim** [jflesch/paperwork#354]
I encounter random segmentation fault while scanning. It occured approximatively 5 times on an hundred scans.

```
…
WARNING paperwork.frontend.util.jobs   Job PageImgLoader:3 took 1121ms and is unstoppable ! (maximum allowed: 500ms)
INFO   paperwork.frontend.util.canvas Animators: 0
INFO   paperwork.backend.docsearch    Updating modified doc: 20141216_0945_49
INFO   paperwork.backend.docsearch    Fitting estimator with doc: 20141216_0945_49 
INFO   paperwork.backend.docsearch    Index: Commiting changes and saving estimators
INFO   paperwork.frontend.mainwindow  Search: 
INFO   paperwork.frontend.util.actions Action: [Open selected document]
INFO   paperwork.frontend.mainwindow  No document selected. Can't open
INFO   paperwork.frontend.util.progressivelist List 'documents' : 100 elements displayed (100 additionnal)
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

INFO   paperwork.frontend.util.actions Action: [Open selected document]
INFO   paperwork.frontend.mainwindow  Showing doc 20141216_0947_40
INFO   paperwork.frontend.mainwindow  Showing document 20141216_0947_40
INFO   root                           Showing page Dummy page
INFO   paperwork.frontend.util.actions Action: [Show a page (selected from the page thumbnail list)]
INFO   paperwork.frontend.util.progressivelist List 'pages' : 0 elements displayed (0 additionnal)
INFO   paperwork.frontend.mainwindow  Image view port resized. (1137, 726) --> (1141, 726)
INFO   root                           Showing page Dummy page
INFO   paperwork.frontend.util.actions Action: [Scan a single page]
INFO   paperwork.frontend.util.config Will scan using hpaio:/net/Photosmart_6520_series?ip=192.168.0.15
INFO   paperwork.frontend.util.config Will scan at a resolution of 300
INFO   paperwork.frontend.util.config Will scan using source Flatbed
INFO   paperwork.frontend.util.config Scanner mode set to 'Color'
INFO   paperwork.frontend.mainwindow  Showing document 20141216_0947_40
INFO   paperwork.frontend.util.canvas Animators: 1
INFO   root                           Showing page Dummy page
INFO   paperwork.frontend.util.progressivelist List 'pages' : 0 elements displayed (0 additionnal)
INFO   paperwork.frontend.mainwindow.scan Scan started
INFO   paperwork.frontend.util.canvas.drawers Drawer: Target area: (307, 0) ((527, 726)) << (307, 0) ((527, 725))
[1]    14619 segmentation fault  paperwork
```

Once it occurs, I can no more scan any document and get this error : 
```
  scanimage -V
scanimage (sane-backends) 1.0.23; backend version 1.0.23
  scanimage -L
device `hpaio:/net/Photosmart_6520_series?ip=192.168.0.15' is a Hewlett-Packard Photosmart_6520_series all-in-one
  scanimage -T
scanimage: open of device hpaio:/net/Photosmart_6520_series?ip=192.168.0.15 failed: Error during device I/O
```

And HP Check fails too :
```
HP-Photosmart-6520-series
-------------------------
Type: Printer
Device URI: hp:/net/Photosmart_6520_series?ip=192.168.0.15
PPD: /etc/cups/ppd/HP-Photosmart-6520-series.ppd
PPD Description: HP Photosmart 6520 Series, hpcups 3.14.3
Printer status: printer HP-Photosmart-6520-series is idle.  enabled since lun. 15 déc. 2014 13:59:51 CET
error: Unable to communicate with device (code=12): hp:/net/Photosmart_6520_series?ip=192.168.0.15
error: unable to open channel
error: Communication status: Failed

Missing Required Dependencies
-----------------------------
None

Missing Optional Dependencies
-----------------------------
None

Total Errors: 1
Total Warnings: 0

Done.
```

The only crappy solution I found for now is to restart my laptop… any idea ?

**Comment by jflesch** [comment]
I can actually reproduce this problem easily : it seems my HP drivers crash every time I scan .. nothing (no paper in the scanner).

**Comment by jflesch** [comment]
I have switched to Debian sid a few months ago, and I haven't been able to reproduce it anymore.
I assume it's a HP driver problem. If so, for the future releases, I intend to release Paperwork using [AppImage](http://appimage.org/). Since appimage include all the required libraries, it should take care of this issue.

**Comment by jflesch** [comment]
The workaround for this ticket will probably also help : https://github.com/jflesch/paperwork/issues/486#issuecomment-235665566
I'm closing this issue. We will see if it's fixed for you too in Paperwork 0.4 when it's released.
